### DISCUSSION TITLE
License collision

### DISCUSSION THREAD

**Discussion by Vadiok** [sachinchoolur/lightGallery#1020] [Q&A]
As I understand from the text of license, if I'm using your gallery at commercial website, I should purchase a commercial license.
But if I fork your library I can publish my fork under the GNU GPL v3 and use this fork in commercial website without purchasing license from you.
Is my explanation correct?

**Comment by sachinchoolur** [comment]
No. Forking an open-source software project does not change the ownership of copyrights in the source code that was formed from the original project. 

From the [GNU docs](https://www.gnu.org/licenses/gpl-faq.html#GPLRequireSourcePostedPublic)
> If you release the **modified version to the public** in some way, the GPL requires you to make the modified source code available to the program's users, under the GPL.

Please read  - https://www.gnu.org/licenses/gpl-3.0.html#section5

Happy to answer if you have any questions.

**Reply by Vadiok** [reply]
Thanks for explanation!
However

> If you release the modified version to the public in some way, the GPL requires you to make the modified source code available to the program's users, under the GPL.

Means, that I cannot change license from GPL to another. There is no information how to handle with other restrictions and modifcations of license.
Your license file says:

> If you are creating an open source application under a license compatible with the GNU GPL license v3, you may use this project under the terms of the GPLv3.

So, following this agreement, if I'll make the fork (which is Open Source), I can release it under the GPLv3 or compatible license which hasn't restrictions on commercial use.

So I think it is more correct to change the license file from two section in one something like:

This software is provided under the license based on GNU GPLv3 except the following cases:
- For commercial usage... (restrictions of commercial usage)
- For any modification versions current license should not be changed.

**Reply by sachinchoolur** [reply]
I think you are confused about the term "commercial" here. 
> If you release the modified version to the public in some way, the GPL requires you to make the modified source code available to the program's users, under the GPL.

It is not just about the commercial use, if you use any GPLv3 licensed software on your website, you'll have to open source the website's source code. I'll try to improve the website docs.

> So, following this agreement, if I'll make the fork (which is Open Source), I can release it under the GPLv3 ~~or compatible license~~ which hasn't restrictions on commercial use.

Of course, you can. But anyone who uses that fork will have to make the source code of the project in which the fork is used, publically available to users.

There are so many projects on GitHub with a similar licensing model. I request you to read the [GPLv3 license agreement](https://www.gnu.org/licenses/gpl-3.0.html) carefully.

If you have any issues with the pricing, please drop me an email at contact@lightgalleryjs.com. I'll try to help you

**Reply by Vadiok** [reply]
Thank you for answering.
Sorry, maybe it is hard for you to understand me, because English is not my native language.

> If you have any issues with the pricing...

No, I just want to understand the license conditions. It is a theoretical question.

> I think you are confused about the term "commercial" here.

As I understand, "commercial" means that is used to receive financial profit. It doesn't matter is the code open or closed. For example if I'll use your library in my website, where I sell something, but the JS code of this site is open, it will means that is a commercial usage?

Am I right or your meaning of word "commercial" differs?  Or the "commercial" means only the license for those who wants to keep the code closed?

> Of course, you can. But anyone who uses that fork will have to make the source code of the project in which the fork is used, publically available to users.

Yes, but not all the code, but the code linked with the library. For example, only JS code, or if it is not used in bundle only the code of fork. So, if the linked source code will be opened, is it possible to use the library in e-commerce?

> There are so many projects on GitHub with a similar licensing model...

Sorry, but I haven't find exactly such types of dual licensing.

The problem is then somebody make the fork under the GPLv3, this fork becomes free for usage in any projects, commercial or not (with some restriction of GPLv3). Or if it is not, it shouldn't be called GPL license.

**Reply by sachinchoolur** [reply]
> Yes, but not all the code, but the code linked with the library. For example, only JS code, or if it is not used in bundle only the code of fork. So, if the linked source code will be opened, is it possible to use the library in e-commerce?

I'm curious to know where did you get this information from.

> c) You must license the **entire work, as a whole,** under this License to anyone who comes into possession of a copy. This License will therefore apply, along with any applicable section 7 additional terms, to the whole of the work, and all its parts, regardless of how they are packaged. This License gives no permission to license the work in any other way, but it does not invalidate such permission if you have separately received it.

👆🏻 From the [GPLv3 license](https://www.gnu.org/licenses/gpl-3.0.html#section5), docs

---
> _The problem is then somebody make the fork under the GPLv3, this fork becomes free for usage in any projects, commercial or not (with some restriction of GPLv3). Or if it is not, it shouldn't be called GPL license._

Using lightGallery or a fork of lightGallery under GPLv3  in your project requires you to release the whole project under the GPLv3. If you are ok with this, please feel free to use lightGallery under GPLv3 without purchasing a commercial license
